lib/state_machine/message.hpp

```diff
@@ -0,0 +1,15 @@
+#pragma once
+
+namespace hyped::state_machine {
+
+// Message class containing messages that prompt transitions
+enum class Message {
+  kNextNominalState,
```

Comment:
How are you going to make sure that receiving two of these in a row is not going to advance the state machine too far? Like, how do you avoid one of the controllers sending this message twice or, more realistically, two individual controllers sending this message and the receiver transitioning twice (instead of the intended once).

Reply:
Was initially decided on to allow different transition orders to end up in the same state, but I guess this doesn't work as well for the NextNominalState transitions as if two messages are sent from different controller to transition from the same state, they would transition too far and essentially skip a state. This would be needed for failure states to ensure that no matter the order of received messages the failure transition is always made. So might be better to add separate messages for all the non-failure transitions, but keep the general transitions for failure states?

Reply:
That may be a good idea, yes.